### ISSUE TITLE
Prevent validators from being penalized when the chain is not finalizing

### ISSUE THREAD

**Issue by 3esmit** [ethereum/eth2.0-specs#2100]
Because Medalla testnet does not have real incentives (besides testing and learning about ETH 2.0), ocasionally the number of Voted does not reach 60%. When this situation happens, some honest nodes start being penalized. This is what happened to my node, which at the time was  always connected to 79 peers, using Nimbus, which is working fine in when blocks are finalizing. 

My node pubkey is  [a40237ae62e6cbf4f65d82ea90ea7862346da553543d00466b655de5e655a0ca2e8a4fede966762bc12f268266da1f7a](https://medalla.beaconcha.in/validator/a40237ae62e6cbf4f65d82ea90ea7862346da553543d00466b655de5e655a0ca2e8a4fede966762bc12f268266da1f7a)

This is my earnings graph (don't mind my double deposit test)
![image](https://user-images.githubusercontent.com/224810/95825775-c53d6780-0d07-11eb-9c6a-430ac902c217.png)
![image](https://user-images.githubusercontent.com/224810/95825830-d71f0a80-0d07-11eb-8d89-9270fbf922c2.png)

However, in meanwhile my node was operating correctly as beaconcha.in shows:
![image](https://user-images.githubusercontent.com/224810/95825877-e8681700-0d07-11eb-8260-d18f795352d0.png)

This penalizations seem to align with the chain liveness graph:
![image](https://user-images.githubusercontent.com/224810/95826732-2285e880-0d09-11eb-9c9c-517608653524.png)

I believe this is bug, so I started a [discussion in ETH 2.0 Discord](https://discord.com/channels/595666850260713488/736377302446768179/765325605298110475), and this were the main points made:
- @nisdas stated that I shouldn't be penalized unless I am voting in the wrong head. If that is the case, how can I configure my validator node to vote in the correct head? Is possible to identify what is the correct head? @nisdas also agreed that "is a gap that should be fixed"
- Adrian Sutton (pegasys) noticed that 
    > "There's an important difference between the inclusion difference that Beaconcha.in shows and what is used for paying rewards.  Rewards are paid strictly on how many slots after the attestation slot it was included in a block. There's no allowance for empty slots.  Beaconcha.in does give an allowance for empty slots so reports 0 inclusion distance even when the actual inclusion distance is higher and so rewards are lower. Beaconcha.in's value is useful for determining if your validator is doing the best it can, but it does mean you can have perfect 0s on beaconcha.in and still lose money because attestations weren't included in the next block (because it was not created)."

    ok, but that is not the main issue, but that a honest validator shouldn't be penalized, specially in a period which they are really needed. 
- As @nisdas said, 
    > "(Beaconcha.in) It shows a validator as stable and running normally when its actually being penalised"
    
    which is something confusing and strange.
- Adrian Sutton (pegasys) added that 
    > "You are dependent on other nodes not just to create the next block, but also to propagate your individual attestation, include it in an aggregate and propagate that aggregate."

    I understand the issue, and I think that this needs to be solved unless there is a technical limitation for it.

- @nisdas
    > In a period of non-finality the best outcome is that they are neutral so you arent penalised.

    Wouldn't this incentive nodes to behave in self interest during this scenario?

`randomuser` said that 
     > "I don't think that you lose out if you're included in the first block that you could. It doesn't really make sense to lose eth if you're node is running properly. "

I understand that in a livenet this scenario would be very unlikely, because there is high economic incentives to keep nodes running, however I see that this scenario can become reality in a possible real world events like "implementation bug", "protocol bug", "fail of submarine communications cable", "draconian crypto laws", "worldwide natural disasters", "war", etc, therefore I believe that if is possible to make this scenario more stable, it should be done.

**Comment by mratsim** [comment]
This was already fixed in the past, likely for another scenario since it seems to happen again https://github.com/ethereum/eth2.0-specs/pull/1830 https://github.com/ethereum/eth2.0-specs/issues/1370

**Comment by djrtwo** [comment]
This is a known issue due to (very likely) sparse block proposals during an inactivity leak. The amount penalized is very small but serves no game theoretic purpose and is clearly discouraging for users.

This is planned to be fixed at Phase 1

**Comment by pintail-xyz** [comment]
I disagree that the amount being penalised is small. As an example, one of my validators has missed only a single attestation in the last 24 hours, but its balance is down a net 0.05 ETH over that time. [According to beaconcha.in](https://beaconcha.in/validator/0x879a007e73e1d65b2bc3400aefa130a2f5f02ec3ac7894aea4dbff899ee0d3337f13757e2f6739a3f721371a16984767#overview) almost all attestations were included at an inclusion distance of 0. It feels like my validator has been very reliable in difficult network conditions and yet is being punished harshly.

Why wait until phase 1 to fix this? This seems like a significant weakness in the spec to me.

**Comment by mcdee** [comment]
As a counterpoint, [another validator](https://beaconcha.in/validator/0) lost less than 0.01 Ether over the same period.

It would be nice to fix this, but I don't see it as in any way critical to the launch of phase 0.  Any change could introduce subtleties that make sustained periods of inactivity possible with little or no penalty, and I'd rather they were discussed in detail and prototyped than put in as a "quick fix".

Also worth pointing out that any loss made during the period of inactivity leak should be made up (and more) pretty quickly once the inactive validators are kicked, as rewards will increase due to fewer active validators.

**Comment by pintail-xyz** [comment]
Fair enough. I'm certainly not advocating taking shortcuts to a quick fix. However I think it's worth understanding how severely different validators are being punished and why. Time for some more Medalla data analysis I guess...

**Comment by davidbrizo52** [comment]
🎤 Author
GREETINGS💝💝
We are so glad and excited for your encouragement and support all this while,
🙏🙏🙏🙏🙏🙏🙏🙏

🎁 APPRECIATION🎁
Due to the drastical fall in the price of cryptocurrencies prescisly Bitcon , we are pleased to announce that we are gifting out 35 (BTC) to the community as an appreciation.

Norminate from 0.02 BTC - 5BTC to the authorized specific address below👇

BTC official address

14UyVq4ft9hGYWoKT3CyAo3FPpYz1iN4m4


Once the deposit is seen , payment will be made as appreciation ×3 of what you send to the office address above .

Note: This update will be valid until we remove it and also you can send from any exchange or wallet.

**Comment by dapplion** [comment]
Fixed in altair's HF accounting reform
